src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -912,13 +912,14 @@ def set_policy(cmd, client, resource_group_name, vault_name,
 
     # Find the existing policy to set
     policy = next((p for p in vault.properties.access_policies
-                   if object_id.lower() == p.object_id.lower() and
+                   if object_id.lower() == p.object_id.lower() and application_id == p.application_id and
```

Comment:
Consider using `lower` to make the comparison case-insensitive.

Reply:
`application_id` could be `None` so I didn't add `lower()`. I was wondering if it's necessary to make it case insensitive since powershell didn't contain such logic and no service ask either.

Reply:
modified